apps/web/app/lib/actions.ts

```diff
@@ -39,6 +39,46 @@ export async function authorizeUser(email: string, password: string) {
   return { success: true };
 }
 
+export async function createCourse(data: CourseFields) {
+  const { subject, code, title } = data;
+  await prisma.course.create({
+    data: {
+      subject,
+      code,
+      title,
+    },
+  });
+
+  return { success: true };
+}
+
+export async function getCourse(subject: string, code: number) {
+  const course = await prisma.course.findFirst({
+    where: { subject: subject, code: code },
+  });
+  return course;
+}
+
+export async function getCourses(subject: string, code?: number) {
+  const courses = await prisma.course.findMany({
+    where: { subject: subject, code: code },
+    orderBy: [{ subject: "asc" }, { code: "asc" }],
+  });
+  return courses;
+}
+
+export async function getAllCourses() {
+  const courses = await prisma.course.findMany({
+    orderBy: [{ subject: "asc" }, { code: "asc" }],
+  });
+  return courses;
+}
+
+export async function deleteCourse(id: string) {
+  await prisma.course.delete({ where: { id } });
+  revalidatePath("/platform/courses");
+}
+
```

Comment:
Server actions emit `POST` requests, meaning these methods are uneccessary and a bad practice. 

When using Next.js Server Actions, the only thing that should be in these files (`actions.ts`) are **mutations**. 

[Server actions are for mutations, not data fetching.](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#authentication-and-authorization:~:text=data%20mutations%20in%20Next.js%20applications.) If you DO want to fetch data, do it directly in the server components or route handlers.